CvGameCoreDLL_Expansion2/CvSpan.h

```diff
@@ -15,7 +15,11 @@
 
 enum DynamicExtentType
 {
+#ifdef __clang__
+	DYNAMIC_EXTENT = INT_MAX,
+#else
 	DYNAMIC_EXTENT = std::size_t(-1),
+#endif // __clang__
```

Comment:
this wasn't necessary for me, are you sure it is required?

Reply:
I was surprised by this as well. ``clang-cl`` completely refused the code when I tried without this. I'll try it again though to see if there's not some other way.

Reply:
Turns out suppressing ``c++11-narrowing`` permits the original code to build. I'm not entirely sure how much I like that though.

Reply:
Ah, I'm using `-w`, which I thought only disabled printing warnings, but now that you mention it I think it might also disable some default `-Werror`. Actually, now that I look closer at this change, I'm not sure that this makes sense. On 64-bit (which I know doesn't actually exist for Civ 5), DynamicExtentType might change size. Since it's not a real enum anyways, this should probably just be `const std::size_t DYNAMIC_EXTENT = -1;`.

Actually actually, https://gcc.godbolt.org/z/qbnY5bjz4 says that `enum { x = std::size_t(-1) }` should work? What error do you get?

Reply:
```
CvGameCoreDLL_Expansion2/CvSpan.h(21,43) : error: non-type template argument evaluates to -1, which cannot be narrowed to type 'std::size_t' (aka 'unsigned int') [-Wc++11-narrowing]
template<typename T, std::size_t TExtent = DYNAMIC_EXTENT>
                                           ^
CvGameCoreDLL_Expansion2/CvSpan.h(118,16) : error: non-type template argument evaluates to -1, which cannot be narrowed to type 'std::size_t' (aka 'unsigned int') [-Wc++11-narrowing]
class CvSpan<T, DYNAMIC_EXTENT>
                ^
CvGameCoreDLL_Expansion2/CvSpan.h(316,61) : error: non-type template argument evaluates to -1, which cannot be narrowed to type 'std::size_t' (aka 'unsigned int') [-Wc++11-narrowing]
inline CvSpan<typename std::iterator_traits<It>::value_type, DYNAMIC_EXTENT> MakeSpan(It first, std::size_t count)
                                                             ^
CvGameCoreDLL_Expansion2/CvSpan.h(321,67) : error: non-type template argument evaluates to -1, which cannot be narrowed to type 'std::size_t' (aka 'unsigned int') [-Wc++11-narrowing]
inline const CvSpan<typename std::iterator_traits<It>::value_type, DYNAMIC_EXTENT> MakeConstSpan(It first, std::size_t count)
                                                                   ^
CvGameCoreDLL_Expansion2/CvSpan.h(327,61) : error: non-type template argument evaluates to -1, which cannot be narrowed to type 'std::size_t' (aka 'unsigned int') [-Wc++11-narrowing]
inline CvSpan<typename std::iterator_traits<It>::value_type, DYNAMIC_EXTENT> MakeSpan(It first, It last)
                                                             ^
CvGameCoreDLL_Expansion2/CvSpan.h(332,67) : error: non-type template argument evaluates to -1, which cannot be narrowed to type 'std::size_t' (aka 'unsigned int') [-Wc++11-narrowing]
inline const CvSpan<typename std::iterator_traits<It>::value_type, DYNAMIC_EXTENT> MakeConstSpan(It first, It last)
                                                                   ^
CvSpanSerialization.h(29,68) : error: non-type template argument evaluates to -1, which cannot be narrowed to type 'std::size_t' (aka 'unsigned int') [-Wc++11-narrowing]
inline FDataStream& operator<<(FDataStream& saveTo, const CvSpan<T, DYNAMIC_EXTENT>& readFrom)
                                                                    ^
CvSpanSerialization.h(38,70) : error: non-type template argument evaluates to -1, which cannot be narrowed to type 'std::size_t' (aka 'unsigned int') [-Wc++11-narrowing]
inline FDataStream& operator>>(FDataStream& loadFrom, const CvSpan<T, DYNAMIC_EXTENT>& writeTo)
                                                                      ^
```

The error is related to its usage as a template parameter. I also observed this working in other instances which is exactly why it surprised me so much. Maybe it's a quirk/bug related to clang for Windows specifically?

Godbolt accepts the following program under ``x86-64 clang`` no problem but attempting to build it with clang on windows leads to the same category of error being reported.

```cpp
#include <cstdlib>
enum DynamicExtentType {
	DYNAMIC_EXTENT = std::size_t(-1),
};

template<typename T, std::size_t TExtent = DYNAMIC_EXTENT>
struct CvSpan {
    std::size_t m;
    CvSpan() : m(TExtent) {}
};

int main(int argc, char** argv) {
    CvSpan<int> span;
}
```

Reply:
Hm, that *is* weird. I bet it's related to delayed template expansion and/or size_t being unsigned int (on Linux it's unsigned long). Does it work with `const std::size_t DYNAMIC_EXTENT = -1;`?

Reply:
``const std::size_t DYNAMIC_EXTENT = -1;`` works like a charm. I'd expected that using a constant in a template argument position would have required `constexpr` so I never ended up trying it.